clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/SerializeApplicationAtomicOperation/SerializeApplicationAtomicOperation.groovy

```diff
@@ -0,0 +1,645 @@
+
+    def json = new JsonBuilder()
+    def applicationMap = [:]
+    applicationMap.resource = resourceMap
+    json(applicationMap)
+    println json.toPrettyString()
+
+    return null
+  }
+
+  private Void initializeResourceMap(Map resourceMap) {
+    resourceMap.google_compute_instance_group_manager = [:]
+    resourceMap.google_compute_instance_template = [:]
+    resourceMap.google_compute_target_pool = [:]
+    resourceMap.google_compute_forwarding_rule = [:]
+    resourceMap.google_compute_http_health_check = [:]
+    resourceMap.google_compute_autoscaler = [:]
+    resourceMap.google_compute_firewall = [:]
+
+    return null
+  }
+
+  /*
+  Need to do this or terraform complains
+   */
+  private Void cleanUpResourceMap(Map resourceMap) {
+    if (numInstanceGroupManagers == 0) {
+      resourceMap.remove("google_compute_instance_group_manager")
+    }
+    if (numInstanceTemplates == 0) {
+      resourceMap.remove("google_compute_instance_template")
+    }
+    if (numTargetPools == 0) {
+      resourceMap.remove("google_compute_target_pool")
+    }
+    if (numForwardingRules == 0) {
+      resourceMap.remove("google_compute_forwarding_rule")
+    }
+    if (numHealthChecks == 0) {
+      resourceMap.remove("google_compute_http_health_check")
+    }
+    if (numAutoscalers == 0) {
+      resourceMap.remove("google_compute_autoscaler")
+    }
+    if (numFirewalls == 0) {
+      resourceMap.remove("google_compute_firewall")
+    }
+  }
+
+  /*
+   * Adds the server group into a resource map that is in Terraform format. Returns false if the server group is invalid.
+   */
+  private Boolean addServerGroupToResourceMap(GoogleServerGroup.View serverGroup, Map resourceMap) {
+
+    def serverGroupMap = [:]
+    if (serverGroup.name) {
+      serverGroupMap.name = serverGroup.name
+      serverGroupMap.base_instance_name = serverGroup.name
+    } else {
+      //required
+      return false
+    }
+    if (serverGroup.zone) {
+      serverGroupMap.zone = serverGroup.zone
+    } else {
+      //required
+      return false
+    }
+
+    if (serverGroup.imageSummary.image) {
+      Map instanceTemplateMap = serverGroup.imageSummary.image
+      InstanceTemplate instanceTemplate = convertMapToInstanceTemplate(instanceTemplateMap)
+      if (!addInstanceTemplateToResourceMap(instanceTemplate, resourceMap)) {
+        return false
+      }
+      serverGroupMap.instance_template = "\${google_compute_instance_template.${instanceTemplate.name}.self_link}"
+    }
+
+    //TODO: get project from account
+    //TODO: named_port, not sure what this is
+    //TODO: update_strategy, automatically restart is default should this be used always (nothing specified in serverGroup object)?
+    if (serverGroup.instanceCounts?.total) {
+      serverGroupMap.target_size = serverGroup.instanceCounts.total
+    }
+
+    if (serverGroup.loadBalancers && !serverGroup.loadBalancers.isEmpty()) {
+      serverGroupMap.target_pools = serverGroup.targetPools
```

Comment:
I think this is the same set of url's you can find by collecting all the google load balancers' `targetPoolUrl`s.